src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Generator.cs

```diff
@@ -123,17 +116,22 @@ public GenerationResult GenerateJsonDocument(SbomPackage packageInfo)
                 ExternalIdentifierType = "purl",
                 Identifier = packageInfo.PackageUrl
             };
-        }
 
-        spdxExternalIdentifier.AddSpdxId();
-        spdxPackage.ExternalIdentifier = new List<string> { spdxExternalIdentifier.SpdxId };
+            spdxExternalIdentifier.AddSpdxId();
+            spdxPackage.ExternalIdentifier = new List<string> { spdxExternalIdentifier.SpdxId };
+        }
```

Comment:
I just noticed that the properties in lined 35-41 throw. Why is that? If those are specific to SPDX2.2, should the `IManifestGenerator` interface be adjusted while we can?

Reply:
Yup this was part of my old PR where we didn't want to introduce any changes to APIs that might be breaking. Since this PR will now be an intentional breaking change, I can update the interface now.